geode-lucene/src/test/java/org/apache/geode/cache/lucene/LuceneQueriesIntegrationTest.java

```diff
@@ -331,6 +331,29 @@ public void queryJsonObject() throws Exception {
   }
 
   @Test()
+  public void waitUntilFlushThrowsIllegalStateExceptionWhenAEQNotFound() throws Exception {
+    Map<String, Analyzer> fields = new HashMap<>();
+    fields.put("name", null);
+    fields.put("lastName", null);
+    fields.put("address", null);
+    luceneService.createIndexFactory().setFields(fields).create(INDEX_NAME, REGION_NAME);
+    Region region = cache.createRegionFactory(RegionShortcut.PARTITION).create(REGION_NAME);
+    final LuceneIndex index = luceneService.getIndex(INDEX_NAME, REGION_NAME);
+
+    // This is to send IllegalStateException from WaitUntilFlushedFunction
+    String nonCreatedIndex = "index2";
+    boolean result = false;
+    try {
+      result = luceneService.waitUntilFlushed(nonCreatedIndex, REGION_NAME, 60000,
+          TimeUnit.MILLISECONDS);
```

Comment:
I think there should be a fail() here,after the all to waitUntilFlushed, if we are expecting an exception to be thrown and it doesn't get thrown.